common/djangoapps/entitlements/helpers.py

```diff
@@ -0,0 +1,63 @@
+from datetime import datetime, timedelta
+
+import pytz
+import sys
+from django.conf import settings
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+def is_entitlement_expired(entitlement):
+    """
+    Determines from the policy if an entitlement can be redeemed, if it has not passed the
+    expiration period of policy.expiration_period_days, and has not already been redeemed
+    """
+    utc = pytz.UTC
+    days_since_created = (datetime.utcnow().replace(tzinfo=utc) - entitlement.created).days
+    expiration_period = settings.ENTITLEMENTS_POLICY.get('expiration_period_days', sys.maxint)
+    if entitlement.policy:
+        expiration_period = entitlement.policy.expiration_period_days
+    return days_since_created > expiration_period and not entitlement.enrollment_course_run
+
+
+def is_entitlement_refundable(entitlement):
+    """
+    Determines from the policy if an entitlement can still be refunded, if the entitlement has not
+    yet been redeemed (enrollment_course_run is NULL) and policy.refund_period_days has not yet passed
+    """
+    utc = pytz.UTC
+    days_since_created = (datetime.utcnow().replace(tzinfo=utc) - entitlement.created).days
+    refund_period = settings.ENTITLEMENTS_POLICY.get('refund_period_days', sys.maxint)
+    if entitlement.policy:
+        refund_period = entitlement.policy.expiration_period_days
+    return (days_since_created < refund_period) and not entitlement.enrollment_course_run
```

Comment:
Will this prevent someone that has spent their entitlement from getting a refund within the first 14 days of a course?

Reply:
Yes it will, if they've redeemed the entitlement they won't be able to redeem anymore according to https://openedx.atlassian.net/wiki/spaces/LEARNER/pages/161761840/Course+Centric+Architecture

Though it won't prevent them from being able to regain that certificate. I guess at that point they would be able to get a refund if we handle that case by unsetting the `enrollment_course_run`